include/multipass/mount_handler.h

```diff
@@ -45,31 +42,62 @@ class MountHandler : private DisabledCopyMove
 
     virtual ~MountHandler() = default;
 
-    // Used to set up anything host side related
-    virtual void init_mount(VirtualMachine* vm, const std::string& target_path, const VMMount& vm_mount) = 0;
-    virtual void start_mount(VirtualMachine* vm, ServerVariant server, const std::string& target_path,
-                             const std::chrono::milliseconds& timeout = std::chrono::minutes(5)) = 0;
-    virtual void stop_mount(const std::string& instance, const std::string& path) = 0;
-    virtual void stop_all_mounts_for_instance(const std::string& instance) = 0;
-    virtual bool has_instance_already_mounted(const std::string& instance, const std::string& path) const = 0;
+    void start(ServerVariant server, std::chrono::milliseconds timeout = std::chrono::minutes(5))
+    {
+        std::lock_guard active_lock{active_mutex};
+        start_impl(server, timeout);
+        active = true;
+    }
```

Comment:
This is the only thing that I am still a little unsure of. The motivation for this scheme was to free implementations from being idempotent (with thread-safety as a bonus). You later said that concrete implementations really ought to be idempotent. OK, but are we sure they are now? I thought you had meanwhile fixed the case that needed that - restart - no? Or are you protecting against other cases where `active` could potentially go out of sync? But if that is a real concern, does it make sense to try to track the state at all? And why the asymmetry with respect to stop? Why do we need `start_impl`s to be idempotent, but not `stop_impl`s?

Reply:
Ahh, yes, I forgot to remove this. Indeed, this was needed by restart, since it didn't stop the mounts before starting them again. But now it does, so `start` can go back to being idempotent.